app/mm-redux/actions/preferences.ts

```diff
@@ -14,28 +14,26 @@ import {PreferenceType} from '@mm-redux/types/preferences';
 import {bindClientFunc} from './helpers';
 import {getProfilesByIds, getProfilesInChannel} from './users';
 import {getChannelAndMyMember, getMyChannelMember} from './channels';
+
 export function deletePreferences(userId: string, preferences: Array<PreferenceType>): ActionFunc {
     return async (dispatch: DispatchFunc, getState: GetStateFunc) => {
         const state = getState();
         const myPreferences = getMyPreferencesSelector(state);
         const currentPreferences = preferences.map((pref) => myPreferences[getPreferenceKey(pref.category, pref.name)]);
 
-        dispatch({
-            type: PreferenceTypes.DELETED_PREFERENCES,
-            data: preferences,
-            meta: {
-                offline: {
-                    effect: () => Client4.deletePreferences(userId, preferences),
-                    commit: {
-                        type: PreferenceTypes.DELETED_PREFERENCES,
-                    },
-                    rollback: {
-                        type: PreferenceTypes.RECEIVED_PREFERENCES,
-                        data: currentPreferences,
-                    },
-                },
-            },
-        });
+        try {
+            dispatch({
+                type: PreferenceTypes.DELETED_PREFERENCES,
+                data: preferences,
+            });
+
+            await Client4.deletePreferences(userId, preferences);
+        } catch {
+            dispatch({
+                type: PreferenceTypes.RECEIVED_PREFERENCES,
+                data: currentPreferences,
```

Comment:
If preference deletion fails for whatever reason, shouldn't the fallback action be to reset with an empty/initial preference set, rather than restoring the existing preferences which is what the user wanted to delete in the first place?

This is pre-existing functionality, so doesn't have to be addressed in this PR.

Reply:
I also may not have a very good handle on what's going on here... so feel free to ignore.

Reply:
You may be deleting only a set of preferences and not everything. Here we are being optimistic and deleting the preferences that you may want to delete (not necessarily all of them), but if it indeed failed then we want to restore the preferences to what you had before deleting them.